pkg/sdk/validation/manifest/json_remote_type.go

```diff
@@ -82,6 +97,55 @@ func (v *RemoteTypeValidator) checkAdditionalRefs(ctx context.Context, entity ty
 	return res, nil
 }
 
+func (v *RemoteTypeValidator) validateBackendStorageSchema(ctx context.Context, entity types.Type) (ValidationResult, error) {
+	res := ValidationResult{}
+
+	if !entity.IsExtendingHubStorage() {
+		return res, nil
+	}
+
+	opts := []public.TypeOption{
+		public.WithTypeFilter(gqlpublicapi.TypeFilter{
+			PathPattern: ptr.String(types.GenericBackendStorageSchemaTypePath),
+		}),
+		public.WithTypeLatestRevision(public.TypeRevisionSpecFields),
+	}
+
+	gotTypes, err := v.hub.ListTypes(ctx, opts...)
+	if err != nil {
+		return res, errors.Wrap(err, "while fetching Types")
+	}
+
+	if len(gotTypes) == 0 || gotTypes[0].LatestRevision == nil || gotTypes[0].LatestRevision.Spec == nil {
+		return res, fmt.Errorf("cannot find generic backend storage schema")
+	}
+
+	storageBytes, err := json.Marshal(gotTypes[0].LatestRevision.Spec.JSONSchema)
+	if err != nil {
+		return res, errors.Wrap(err, "while marshaling generic backend storage schema to bytes")
+	}
+
+	var storageSchema string
+	err = json.Unmarshal(storageBytes, &storageSchema)
+	if err != nil {
+		return res, errors.Wrap(err, "while converting generic backend storage schema to string")
+	}
+
+	schemaLoader := gojsonschema.NewStringLoader(storageSchema)
```

Comment:
You don't need to do that - you can simply use different loader:

```suggestion
	schemaLoader := gojsonschema.NewGoLoader(gotTypes[0].LatestRevision.Spec.JSONSchema)
```

But then you need to update tests:

```go
func fixGQLTypeSchema(t *testing.T, path, schema string) *gqlpublicapi.Type {
	var schemaObj interface{}
	err := json.Unmarshal([]byte(schema), &schemaObj)
	require.NoError(t, err)
	return &gqlpublicapi.Type{
		Path: path,
		LatestRevision: &gqlpublicapi.TypeRevision{
			Spec: &gqlpublicapi.TypeSpec{
				JSONSchema: schemaObj,
			},
		},
	}
}
```

Reply:
I tested it and it is not so clear. The `NewGoLoader` requires that the `interface{}` will be a Go struct and in our `hub-manifests` we are using string 
```
spec:
  jsonSchema:
    value: |-
      {
        "$schema": "http://json-schema.org/draft-07/schema",
        "type": "object",
```

We could presumably toggle this but I am not sure if we want to invest time in improving this.

Reply:
So if this is string, why do you marshal it and umarshal again? 🙂  What I provided you is a ready-to-copy snippet which works well. Go loader underneath does what you do - so if we can reuse something which is already there, why not to do that? In a result we have simplified, easier-to-read code.